WordPress/gutenberg
# Template Part: Swap block action places

## What?
Resolves #41437.
Resolves #34737.

PR swaps places for "Edit" and "Replace" template part actions.

## Why?
The "Edit" should be the primary action for the Template Part block.

## How?
The Site Editor now injects the "Edit" action via a filter. We use a similar method for the Widgets screen to [inject the "Move to" action](https://github.com/WordPress/gutenberg/blob/trunk/packages/edit-widgets/src/filters/move-to-widget-area.js). I've removed old component.

The "Replace" action is now uses `BlockSettingsMenuControls` SlotFill, directly block block edit component. I've not seen this slot used directly from a block before (at least in core), so I would love to hear your opinion on this one 🙇 

## Testing Instructions
1. Open the Site Editor.
2. Open any template with parts.
3. Confirm both "Edit" and "Replace" actions work as expected.
4. Test with Template Parts with footer or header area assigned; they display "Replace" action.
5. Template Part in placeholder state shouldn't display these actions.

## Screenshots or screencast 

https://user-images.githubusercontent.com/240569/177741501-7e208699-7cf9-4813-95be-22185b53dc60.mp4
